### DISCUSSION TITLE
Syntax Highlighting

### DISCUSSION THREAD

**Discussion by sebkolind** [jesseduffield/lazygit#1335] [Q&A] [answered]
Is it somehow possible to get syntax highlighting on the code in the pane to the right (default layout)? It would be a big improvement when solving big merge conflicts, and in general it just looks better.

For me, this is the only thing lazygit lacks to be the perfect substitution for pretty much any GUI git client I have tried.

**Comment by mjarkk** [comment, marked as answer]
You can use a git diff viewer like [delta](https://github.com/dandavison/delta) to get syntax highlighting.  
See this issue for how other people have configured it in lazygit: https://github.com/jesseduffield/lazygit/issues/893

**Reply by sebkolind** [reply]
Thanks. It seems to work, however if I enter a changed file (using Enter) it is not using Delta anymore. Do you know if this is how it should be, or I need to setup up differently? I used the configuration from [here](https://github.com/jesseduffield/lazygit/blob/master/docs/Custom_Pagers.md#delta)

**Reply by mjarkk** [reply]
Dunno @jesseduffield maybe knows how this works

**Reply by jesseduffield** [reply]
The reason for that is so that we can allow you to stage a file line by line. Currently, our line-by-line staging process can only work with plain git diffs. Supporting each external git pager (e.g. Delta) would be difficult

**Reply by danny-does-stuff** [reply]
I was surprised by this as well. I learned you can focus the unstaged changes (while still using delta) by just hitting 0 to focus that pane. 

Thanks for the awesome library, just got it last week and I <3 it!

**Comment by prabanjanraja** [comment]
I have configured delta for syntax highlighting and it works fine with the gitdiff command but even after updating the config.yml file it does not gets reflected in lazygit's diff windows
This is the configuration I used
```
git:
  paging:
    colorArg: always
    useConfig: true
```
![image](https://github.com/jesseduffield/lazygit/assets/56388091/48385fa0-60a7-4abb-91f6-4c11e7203789)
![image](https://github.com/jesseduffield/lazygit/assets/56388091/47ee0a77-ff75-4f3e-82e0-e1f2e4b8cdc8)
